owl-close-world/src/main/java/io/opencaesar/closeworld/ClassExpression.java

```diff
@@ -254,20 +254,23 @@ protected String toAtom() {
 
 	/**
 	 * Singleton implements methods for ClassExpressions that encapsulate an arbitrary
-	 * object representing a single class.
+	 * object representing a single OWL2-DL class.
```

Comment:
ClassExpression.java and Taxonomy.java are not specific to OWL. The internals of the class are opaque to the code. If you expect to serialize the resulting closure to OWL, then the classes need to be OWL classes. The algorithm itself does not require that.

Moreover, there's no such thing as and OWL2-DL class. An OWL class is an OWL class. Whether an ontology can be classified as DL depends on its axioms, not the nature of its classes.